libqtile/resources/default_config.py

```diff
@@ -80,6 +80,12 @@
     Key([mod], "r", lazy.spawncmd(), desc="Spawn a command using a prompt widget"),
 ]
 
+if qtile.core.name is "wayland":
+    for i in range(1,7):
+        keys.extend([
+            Key(["control", "mod1"], f"f{i}", lazy.core.change_vt(i), desc=f"Switch to VT{i}")
+        ])
```

Comment:
Sadly when the default config is imported, `libqtile.qtile` is `None`, so we can't have backend checks in the default config.

One option would be to comment it out, but I suspect we specifically want this one to be enabled.

Another option would be to enable the binding even on X, as it would be a no-op anyway. In which case, we'd need to be careful not to override the existing VT-switching bindings of the X server.

I suggest we do add bindings to any backend, but we change the keys:
```suggestion
for vt in range(1, 7):
    keys.append(
	Key(["mod1"], f"f{vt}", lazy.core.change_vt(vt), desc=f"Switch to VT{i}")
    )
```

Reply:
Could we extend the `when()` method for lazy calls to only run on specific backends?

Reply:
Interesting. Yeah, that could be cool, and might nicely give when() some more visibility. +1 for that

Reply:
It wouldn't stop the key being bound unless we added code to check if the backend had been specified and remove the binding when the backend doesn't match.

Reply:
Isn't the key replayed if the when() failed?

Reply:
This shouldn't be a problem anymore since we merged #4544, right? If so, let's merge it as is to relieve @cjbayliss and open another PR if we want to change the code to something we consider better (e.g. using the new `when()` arguments).